ec/ecresource/deploymentresource/deployment/v2/schema.go

```diff
@@ -111,7 +108,7 @@ func DeploymentSchema() schema.Schema {
 				Sensitive: true,
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.UseStateForUnknown(),
-					useNullIfNotAPM{},
+					UseNullUnlessAddingAPMOrIntegrationsServer(),
```

Comment:
What will happen when APM or IntegrationsServer are deleted from configuration?

Reply:
In that case, we retain the existing secret token. The token is only returned the  _first_ time a resource is enabled, i.e:

1. Enable integrations server -> Token returned
2. Disable integrations server -> No token obviously
3. Re-enable integrations server -> Still no token, since it was set in 1. 

In reality this is a cause for bugs, but that's how the Cloud API works right now.